views/default/input/submit.php

```diff
@@ -5,10 +5,16 @@
  * @package Elgg
  * @subpackage Core
  *
- * @uses $vars['class'] CSS class that replaces elgg-button-submit
+ * @uses $vars['class'] Additional CSS class
  */
 
 $vars['type'] = 'submit';
-$vars['class'] = elgg_extract('class', $vars, 'elgg-button-submit');
+
+if (!isset($vars['class'])) {
+	$vars['class'] = array('elgg-button-submit');
+} elseif (!is_array($vars['class'])) {
+	$vars['class'] = array($vars['class']);
+	$vars['class'][] = 'elgg-button-submit';
+}
```

Comment:
> We could also make this happen in elgg_format_attributes(), after all noone needs duplication in 'class' attribute. @mrclay ?

i also thought of that, but then we need to move a lot of intelligence (like exploding and making it unique) in that function. Do we want that?



> I don't think array unique will work. Someone would have passed it as string 'elgg-button elgg-button-submit elgg-button-something' to vars, so I think we need to collapse the final class string before running through array_unique...

then we should explode the class string first...